iseq.c

```diff
@@ -928,6 +927,18 @@ rb_iseq_new_main(const rb_ast_body_t *ast, VALUE path, VALUE realpath, const rb_
 rb_iseq_t *
 rb_iseq_new_eval(const rb_ast_body_t *ast, VALUE name, VALUE path, VALUE realpath, VALUE first_lineno, const rb_iseq_t *parent, int isolated_depth)
 {
+    VALUE coverages = rb_get_coverages();
+    if (RTEST(coverages) && RTEST(path) && !RTEST(rb_hash_has_key(coverages, path))) {
+        int line_offset = RB_NUM2INT(first_lineno) - 1;
+        int line_count = line_offset + ast_line_count(ast);
+
+        if (line_count >= 0) {
```

Comment:
@ioquatix Sorry to review after the merge, but this part is almost a code clone of the beginning of rb_iseq_new_top. Can you refactor it out to a function?

Reply:
Sure, no problem.